sdks/v4-sdk/src/PositionManager.ts

```diff
@@ -284,9 +282,11 @@ export abstract class V4PositionManager {
 
     let value: string = toHex(0)
 
+    let needToSendEth = isMint(options) && options.migrateOptions?.neededCurrency?.isNative
```

Comment:
its weird to me that `migrateOptions.migrate` could be `false`, but then `neededCurrency` could still be set?
By definition if `migrateOptions.neededCurrency` is set, you're migrating no? It feels like either `migrate` should be a boolean OUTSIDE of the options, and then if `migrate`, you check `options`.
OR you should have an invariant that says if `neededCurrency` is set, `migrate` must be true